Fix inconsistent function protocol types

ricfehr3 opened the pull request:
Fixes function prototypes in bigmap.c being inconsistent to implementations, causing issues when types change. This change makes it compatible with Tonc v1.4.5.

AntonioND commented:
I think the definitions are fine, it's the prototypes the ones that are wrong. If you check `main()` you can see the functions receive integers without their fraction, not fixed point values:

```
x= g_link.x>>8, y= g_link.y>>8;

vp_center(&g_vp, x, y);
```

In fact, you can delete the prototypes (and reorder `vp_set_pos()` and `vp_center()`) so that we don't have this issue anymore.

ricfehr3 commented:
I gotcha. I updated the PR to reflect that. I also added a change to use the tonc function `fx2int` everywhere. I think this would be better since, A: It's not so obvious that it's converting for fixed-point, and B: It leaves precision up to tonc.

ricfehr3 commented:
Follow up question. Is there any harm to only using types provided by tonc? I don't think it's necessary, but I think the consistency would be nice style wise.

AntonioND commented:
Thanks! This certainly makes some things clearer.

Regarding the types, it depends on who you ask. I personally prefer to use `int` and similar types because that's what the standard library uses, but some libraries may want to use their own types internally in case they are ported to some platform that breaks some assumption.
